alacritty/src/display/mod.rs

```diff
@@ -1331,16 +1339,25 @@ impl Display {
     }
 
     /// Check whether a hint highlight needs to be cleared.
-    fn validate_hints(&mut self, display_offset: usize) {
+    fn validate_hint_highlights(&mut self, display_offset: usize) {
         let frame = self.damage_tracker.frame();
-        for (hint, reset_mouse) in
-            [(&mut self.highlighted_hint, true), (&mut self.vi_highlighted_hint, false)]
-        {
+        let hints = [
+            (&mut self.highlighted_hint, &mut self.highlighted_hint_age, true),
+            (&mut self.vi_highlighted_hint, &mut self.vi_highlighted_hint_age, false),
+        ];
+        for (hint, hint_age, reset_mouse) in hints {
             let (start, end) = match hint {
                 Some(hint) => (*hint.bounds().start(), *hint.bounds().end()),
-                None => return,
+                None => continue,
             };
 
+            *hint_age += 1;
+
+            // Ignore hints that were created this frame.
+            if *hint_age == 1 {
+                continue;
+            }
```

Comment:
My initial plan was to fix search invalidating the hint by re-validating the hint on the second frame, but not re-validating on the third frame. However turns out that just not invalidating it on the first frame already does the job.

I still think it's best to keep this as an integer, because I can definitely see the concept of re-validating it on the second frame coming into play again with possible future issues.